Rakefile

```diff
@@ -6,31 +6,59 @@ namespace :book do
     end
   end
 
+  # Variables referenced for build
+  version_string = ENV['TRAVIS_TAG'] || `git describe --tags`.chomp
+  if version_string.empty?
+    version_string = '0'
+  end
+  date_string = Time.now.strftime("%Y-%m-%d")
+  params = "--attribute revnumber='#{version_string}' --attribute revdate='#{date_string}'"
+
   desc 'build basic book formats'
-  task :build do
+  task :build => [:build_html, :build_epub, :build_pdf] do
+    begin
+        # Run check
+        Rake::Task["book:check"].invoke
+
+        # Rescue to ignore checking errors
+        rescue => e
+        puts e.message
+        puts "Error when checking books (ignored)"
+    end
+  end
 
+  desc 'build basic book formats (for ci)'
+  task :ci => [:build_html, :build_epub, :build_pdf] do
     begin
-      version_string = ENV['TRAVIS_TAG'] || `git describe --tags`.chomp
-      if version_string.empty?
-        version_string = '0'
-      end
-      date_string = Time.now.strftime("%Y-%m-%d")
-      params = "--attribute revnumber='#{version_string}' --attribute revdate='#{date_string}'"
+        # Run check, but don't ignore any errors
+        Rake::Task["book:check"].invoke
+    end
+  end
+
+  desc 'generate contributors list'
+  file 'book/contributors.txt' do
```

Comment:
@jnavila do you mean comparing timestamp of `contributors.txt` against generated books (such as `progit.html`)?

Reply:
The file is generated from the git history, so the timestamps are not really useful. Also, the contributors file is regenerated before progit.html, so it will always trigger.

Reply:
@jnavila so what should i compare then between `contributors.txt` and output files (such as `progit.html`) if comparing timestamp is useless and not an option?

Reply:
Good question. I would find a way to include the git hash at the time of the generation, and compare it to the actual HEAD. If it's too cumbersome, be less smart and generate all the time.